contracts/contracts/vault/VaultCore.sol

```diff
@@ -165,20 +165,21 @@ contract VaultCore is VaultStorage {
         emit Redeem(msg.sender, _amount);
 
         // Send outputs
-        for (uint256 i = 0; i < allAssets.length; i++) {
+        uint256 assetCount = allAssets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
             if (outputs[i] == 0) continue;
 
-            IERC20 asset = IERC20(allAssets[i]);
+            address assetAddr = allAssets[i];
 
-            if (asset.balanceOf(address(this)) >= outputs[i]) {
+            if (IERC20(assetAddr).balanceOf(address(this)) >= outputs[i]) {
                 // Use Vault funds first if sufficient
-                asset.safeTransfer(msg.sender, outputs[i]);
+                IERC20(assetAddr).safeTransfer(msg.sender, outputs[i]);
             } else {
-                address strategyAddr = assetDefaultStrategies[allAssets[i]];
+                address strategyAddr = assetDefaultStrategies[assetAddr];
```

Comment:
interesting, I hadn't realized that reading an int in a fixed-length array would significantly change the gas usage

Reply:
Yes, this will do an SLOAD so best to cache in memory if used more than once.
Reading the array length is also a SLOAD

Reply:
got it thanks